core/main.go

```diff
@@ -86,5 +120,8 @@ func main() {
 	}()
 
 	<-xsignal.WaitForInterrupt()
+	if err := stopRunningServices(dep.api); err != nil {
+		logrus.Fatalln(err)
+	}
 	server.Close()
```

Comment:
closing gRPC server first is better otherwise some services can be started by devs during the stopping process

Reply:
good point but at the same time these services rely on the api so to gracefully shutdown the api we need to close all the connections first. If we don't the `server.Close()` is waiting for it and never stops.

Reply:
I agree with you @antho1404.
The perfect way will be do "deactivate" the api service start but I think it's too much for now.

Reply:
How about closing the [listener](https://github.com/mesg-foundation/core/blob/62bf84c/interface/grpc/server.go#L51)? That way grpc pkg will not be able to _Accept()_ new connections but currently accepted ones still be active.

I moved this to a new [issue](https://github.com/mesg-foundation/core/issues/803), please resolve this conversation after you read.